kratos/utilities/parallel_levelset_distance_calculator.h

```diff
@@ -148,153 +151,116 @@ class ParallelDistanceCalculator
      * @param max_distance distances will not be computed after reaching this limit
      * @see ParallelDistanceCalculator::CalculateDistances
      */
-    void CalculateDistancesLagrangianSurface(ModelPart& rModelPart,
-            const Variable<double>& rDistanceVar,
-            const Variable<double>& rAreaVar,
-            const unsigned int max_levels,
-            const double max_distance)
+    void CalculateDistancesLagrangianSurface(
+        ModelPart& rModelPart,
+        const Variable<double>& rDistanceVar,
+        const Variable<double>& rAreaVar,
+        const unsigned int MaxLevels,
+        const double MaxDistance)
     {
         KRATOS_TRY
         bool is_distributed = false;
         if(rModelPart.GetCommunicator().TotalProcesses() > 1)
             is_distributed = true;
 
-        array_1d<double,TDim+1> visited;
-        const int elem_size = rModelPart.Elements().size();
-        const int node_size = rModelPart.Nodes().size();
-
         //check that variables needed are in the model part
-        if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rDistanceVar)) )
-            KRATOS_THROW_ERROR(std::logic_error,"distance Variable is not in the model part","");
-        if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rAreaVar)) )
-            KRATOS_THROW_ERROR(std::logic_error,"Area Variable is not in the model part","");
-
+        const int node_size = rModelPart.Nodes().size();
+        KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rDistanceVar))) << "Distance variable is not in the model part" << std::endl;
+        KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rAreaVar))) << "Area Variable is not in the model part" << std::endl;
```

Comment:
by now you can also check the ModelPart directly
```suggestion
        const int node_size = rModelPart.Nodes().size();
        KRATOS_ERROR_IF_NOT(rModelPart.HasNodalSolutionStepsVariable(rDistanceVar)) << "Distance variable is not in the model part" << std::endl;
        KRATOS_ERROR_IF_NOT(rModelPart.HasNodalSolutionStepsVariable(rAreaVar)) << "Area Variable is not in the model part" << std::endl;
```